### PR TITLE
Add remaining DCW questionnaires

### PR DESCRIPTION
**Pull request by mcjustin**
Per https://github.com/uwcirg/asbi-screening-app/issues/34

### CODE REVIEW

File: src/fhir/1_Questionnaire-C-IDAS.json [hunk 1]
```diff
@@ -0,0 +1,1433 @@
...
+            "display": "Unable to evaluate"
+          }
+        }
+      ]
+    },
+    {
+      "type": "choice",
+      "required": false,
+      "linkId": "cs-idas-7",
+      "text": "Loss of interest - less involved in usual activities",
```
Comment (sierramatice) [inline src/fhir/1_Questionnaire-C-IDAS.json:476]: There are a few minor differences between this assessment and the PDF they provided, but if this was the one they identified in the FHIR library I think we can dismiss those differences

File: src/fhir/1_Questionnaire-GDS.json [hunk 2]
```diff
@@ -0,0 +1,745 @@
...
+      ],
+      "required": false,
+      "linkId": "/48512-8",
+      "text": "Are you basically satisfied with your life",
+      "answerOption": [
+        {
+          "extension": [
+            {
+              "url": "http://hl7.org/fhir/StructureDefinition/questionnaire-optionPrefix",
+              "valueString": "1"
```
Comment (sierramatice) [inline src/fhir/1_Questionnaire-GDS.json:40]: The scoring for this one needs to be updated, inconveniently it is not always 1 pt for yes, 0 points for no

Comment (sierramatice) [reply src/fhir/1_Questionnaire-GDS.json:40]: @achen2401 I am happy to go in and fix if you would like me to!

File: src/fhir/1_Questionnaire_CP-ECOG.json [hunk 3]
```diff
@@ -0,0 +1,1161 @@
+{
+  "resourceType": "Questionnaire",
+  "id": "CIRG-CP-ECOG",
+  "name": "CP-ECOG",
+  "title": "Everyday Cognition Care Partner Assessment (ECog-12 Care Partner)",
+  "status": "active",
+  "item": [
+    {
```
Comment (sierramatice) [inline src/fhir/1_Questionnaire_CP-ECOG.json:8]: This assessment is missing the first question - Are you concerned that your care recipient has a memory or other thinking problem? Y/N

File: src/fhir/1_Questionnaire-GDS.json (hunk 2 above)
Comment (achen2401) [reply src/fhir/1_Questionnaire-GDS.json:40]: @sierramatice The scoring is accounted for [here](https://github.com/uwcirg/asbi-screening-app/pull/35/files#diff-f79a90b783ca95682f6a089e00fb2a2284b12207f2f383b60b245034dc22b643)  line 187 to 295.  Will you check there to see if the score(s) are calculated as intended (some are given 1 points for yes, some are given 1 point for answer no)

**Review by sierramatice — CHANGES_REQUESTED** [review]
@achen2401 I found a few small things missing or that need tweaking, and added comments in the code. Let me know if you need more details!

File: src/fhir/1_Questionnaire-GDS.json (hunk 2 above)
Comment (sierramatice) [reply src/fhir/1_Questionnaire-GDS.json:40]: @achen2401 Looks fantastic, Sorry!

File: src/fhir/1_Questionnaire_CP-ECOG.json (hunk 3 above)
Comment (achen2401) [reply src/fhir/1_Questionnaire_CP-ECOG.json:8]: thank you @sierramatice I added the missing question

File: src/fhir/1_Questionnaire-C-IDAS.json (hunk 1 above)
Comment (achen2401) [reply src/fhir/1_Questionnaire-C-IDAS.json]: thank you @sierramatice I added the missing text from the PDF they provided.

**Comment by achen2401** [comment]
@sierramatice I fixed the issues you found.  Will you review again when you have a moment.

File: src/fhir/1_Questionnaire_CP-ECOG.json (hunk 3 above)
Comment (sierramatice) [reply src/fhir/1_Questionnaire_CP-ECOG.json:8]: @achen2401 we should keep the title of the ECog-12 Care Partner in the same format as the title of the ECog-12

**Review by sierramatice — COMMENTED** [review]
@achen2401, I added a comment about the title of "Everyday Cognition Care Partner Assessment (for Care Partner)",

File: src/fhir/1_Questionnaire_CP-ECOG.json (hunk 3 above)
Comment (achen2401) [reply src/fhir/1_Questionnaire_CP-ECOG.json:8]: @sierramatice thanks.  I changed the title to `Everyday Cognition Care Partner Assessment (ECog-12 Care Partner)` to match how it is being referenced in the introduction text.  I hope this works?

**Comment by sierramatice** [comment]
Looks great Thank you!

**Review by sierramatice — APPROVED** [review]

**Comment by mcjustin** [comment]
@achen2401 are there any blockers to rolling this out? I can add a few representative CarePlans (and Patients) to this in a subsequent MR.

**Comment by achen2401** [comment]
@mcjustin Will you have a moment to review these changes?  It will be good to have your eyes on these.

**Comment by mcjustin** [comment]
> @mcjustin Will you have a moment to review these changes? It will be good to have your eyes on these.

Thanks for flagging, I hadn't noticed some of your library / cql work. Reviewing in more depth now.

File: src/cql/upstream-external/C-IDAS_LogicLibrary.cql [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+    (getCurrentResponseByQuestionId(question16LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question17LinkId)) is not null or
+    (getCurrentResponseByQuestionId(question18LinkId)) is not null
+
+   ) // at least one response for all the questions
+    and DisplayScreeningScores // Input parameter which allows score display to be disabled
+
+// Calculate the score for the Full C-IDAS
+// sum up all numeric values for all the answers
+define ScoreFullCIDAS:
```
Comment (mcjustin) [inline src/cql/upstream-external/C-IDAS_LogicLibrary.cql:335]: As we discussed just now, much of the scoring code is repeated in https://github.com/uwcirg/patient-summary/tree/feature/initial-patient-summary , would be good to move this into a Library. We also discussed sharing a Library w/ COSRI, and the fact that COSRI's cql-execution would need to be updated (currently uses 1.3.7, while this repo uses 2.4.x)

**Comment by mcjustin** [comment]
Noting that we don't currently implement cross-questionnaire conditions, only within a questionnaire (and, this PR doesn't add that, which is fine)

**Comment by achen2401** [comment]
@mcjustin Just FYI, the branch should now have the latest from the master branch.

**Comment by mcjustin** [comment]
@sierramatice  these are now available at https://femr.dcw.dev.cirg.uw.edu ; the conditions aren't implemented, but can you please review for questionnaire content and scoring? Perhaps a phased approach would be good, where you do a quick review and if it looks good we'll merge, and then later a more in-depth set of tests and record them in a test matrix and we can use a new pull request to address any changes needed.
@achen2401 fyi per our convo earlier.

**Comment by mcjustin** [comment]
@achen2401  I get "Unsupported item type" for behav5

**Comment by achen2401** [comment]
> I get "Unsupported item type" for behav5
@mcjustin I didn't get the error when I tried in my instance.  Can you tell if the error is coming from the `questionnaire-to-survey` tool library we are using?

**Comment by mcjustin** [comment]
> > I get "Unsupported item type" for behav5
> > @mcjustin I didn't get the error when I tried in my instance.  Can you tell if the error is coming from the `questionnaire-to-survey` tool library we are using?

@achen2401  just reproduced the issue:
```
loading instrument css error: Error: Cannot find module './behav5.scss'
```
`XHR GET https://fhirwall.dcw.dev.cirg.uw.edu/fhir/Questionnaire?name:contains=behav5`
`Error loading Questionnaire  Error: Unsupported item type.`

stack includes `index.[some file build hex code?].js` and `chunk-vendors[different hex code].js`

**Comment by achen2401** [comment]
@mcjustin Please see my fix [here](https://github.com/uwcirg/asbi-screening-app/pull/35/commits/70eb2228e6696265a9e0f1c52d61507fa9d92f5e) Here is the screenshot of where the error is:
Screen Shot 2022-11-08 at 4 57 55 PM

You will need to r**e-upload** the modified behav5 questionnaire to the FHIR host server before trying again

**Pull request merged**